File: moneybox-server/src/main/java/com/samoshin/config/JsonErrorHandlingDeserializer.java
package com.samoshin.config;

import org.apache.kafka.common.serialization.Deserializer;
import org.springframework.kafka.support.serializer.ErrorHandlingDeserializer;
import org.springframework.kafka.support.serializer.JsonDeserializer;

public class JsonErrorHandlingDeserializer<T> implements Deserializer<T> {

    private final ErrorHandlingDeserializer<T> errorHandlingDeserializer;

    public JsonErrorHandlingDeserializer(Class<T> targetType) {
        JsonDeserializer<T> jsonDeserializer = new JsonDeserializer<>(targetType);
        jsonDeserializer.addTrustedPackages("*");
        this.errorHandlingDeserializer = new ErrorHandlingDeserializer<>(jsonDeserializer);
    }

    @Override
    public T deserialize(String topic, byte[] data) {
        return errorHandlingDeserializer.deserialize(topic, data);
    }
}

File: moneybox-server/src/main/java/com/samoshin/config/JsonErrorHandlingDeserializer.kt
package com.samoshin.config

import org.apache.kafka.common.serialization.Deserializer
import org.springframework.kafka.support.serializer.ErrorHandlingDeserializer
import org.springframework.kafka.support.serializer.JsonDeserializer

class JsonErrorHandlingDeserializer<T>(targetType: Class<T>?) : Deserializer<T> {
    private val errorHandlingDeserializer: ErrorHandlingDeserializer<T>

    init {
        val jsonDeserializer = JsonDeserializer(targetType)
        jsonDeserializer.addTrustedPackages("*")
        this.errorHandlingDeserializer = ErrorHandlingDeserializer(jsonDeserializer)
    }

    override fun deserialize(topic: String, data: ByteArray): T {
        return errorHandlingDeserializer.deserialize(topic, data)
    }
}